include/grpc/impl/codegen/grpc_types.h

```diff
@@ -298,8 +298,12 @@ typedef enum grpc_call_error {
 /** Force compression to be disabled for a particular write
     (start_write/add_metadata). Illegal on invoke/accept. */
 #define GRPC_WRITE_NO_COMPRESS (0x00000002u)
+/** Force coalescing of last streaming message and trailing metadata into the
+   same core batch */
+#define GRPC_WRITE_LAST_MESSAGE (0x00000004u)
```

Comment:
This probably shouldn't be added to the core API: it's a C++ level concept (that just triggers sending an additional batch).

Instead, add a bool to WriteOptions